django_password_validators/password_history/password_validation.py

```diff
@@ -45,15 +48,32 @@ def validate(self, password, user=None):
 
         for user_config in UserPasswordHistoryConfig.objects.filter(user=user):
             password_hash = user_config.make_password_hash(password)
+
             try:
-                PasswordHistory.objects.get(
-                    user_config=user_config,
-                    password=password_hash
-                )
-                raise ValidationError(
-                    _("You can not use a password that is already used in this application."),
-                    code='password_used'
-                )
+                if self.lookup_range == 0:
+                    raise PasswordHistory.DoesNotExist
+
+                current_user_passwords = PasswordHistory.objects.filter(
+                    user_config=user_config).order_by('date')
+
+                if self.lookup_range >= len(current_user_passwords):
+                    PasswordHistory.objects.get(
+                        user_config=user_config,
+                        password=password_hash
+                    )
```

Comment:
@spokerman12, if the above query executes successfully, it means that there is an existing password that matches the new password and hence the validation error should be raised as it was done before the changes in this PR.

Reply:
That's correct, but note that the `validate` method represents: `if this is a valid password, we raise PasswordHistory.DoesNotExist else do nothing or raise a different exception`

Reply:
OK, my comment was wrong. Addressing the errors.

Reply:
OK, I had an important error regarding this comment. Addressing right now

Reply:
Yep, that's right; I had a logic error. It's fixed now, will address the line in this conversation.